[2.7.0] filter-tenants only covers syntax errors

#### Setup Details
 CIS Version : 2.7.0    
 Build: f5networks/k8s-bigip-ctlr:latest       
BIGIP Version: Big IP 15.1.0.5
AS3 Version: 3.22.1
Agent Mode: AS3
Orchestration: K8S
Orchestration Version: 1.21.5
Pool Mode: Nodeport
Additional Setup details: Calico

#### Description  
  
Need to be able to allow configmaps to be pushed to BIG-IP even in the case where a bad configmap has bad configuration, not just syntax.

#### Steps To Reproduce
1) Deploy configmap with a template type that does not exist to simulate a bad configmap (see example below)
2) Confirm error in controller logs
3) Attempt to deploy another working configmap without any errors, and see that it does not get deployed (blocked)

#### Expected Result 

configmaps with errors should not block other working configmaps when using filter-tenants=true

#### Actual Result  

configmaps are blocked when there is a semantic error

#### Diagnostic Information

bad configmap (contains a template that does not exist..    ` "template": "http3"` )
```
kind: ConfigMap
apiVersion: v1
metadata:
  name: f5-as3-declaration5
  namespace: default
  labels:
    f5type: virtual-server
    as3: "true"
data:
  template: |
    {
        "class": "AS3",
        "declaration": {
            "class": "ADC",
            "schemaVersion": "3.20.0",
            "id": "urn:uuid:5",
            "label": "HTTPS VS with WAF",
            "remark": "CICD example",
            "Tenant5": {
                "class": "Tenant",
                "Test1": {
                    "class": "Application",
                    "template": "http3"
                    "san_test_svcaddr": {
                        "class": "Service_Address",
                        "trafficGroup": "/Common/traffic-group-1",
                        "virtualAddress": "200a:b:c:d:e::5006"
                    },
                    "serviceMain": {
                        "class": "Service_HTTP",
                        "virtualAddresses": [{ "use": "san_test_svcaddr"}],
                        "pool": "f5_pool"
                    },
                    "f5_pool": {
                        "class": "Pool",
                         "monitors": [ {
                           "use": "http_mon"
                         }],
                        "members": [
                            {
                                "servicePort": 80,
                                "serverAddresses": [],
                                "shareNodes": true
                            }
                        ]
                    },
                    "http_mon" : {
                      "class": "Monitor",
                      "monitorType": "http",
                      "send": "GET /apim HTTP/1.1\r\nHost: mytest.co.jp\r\nConnection: Close",
                      "receive": "404",
                      "targetPort": 8080
                  }
                }
            }
        }
    }
```

Confirm the errors

```
2022/01/31 06:29:18 [DEBUG] [AS3] posting request to https://172.8.1.147/mgmt/shared/appsvcs/declare/Tenant5
2022/01/31 06:29:18 [ERROR] [AS3] Big-IP Responded with code: 422
2022/01/31 06:29:18 [ERROR] [AS3] Raw response from Big-IP: map[code:422 message:Invalid data property: http3]
2022/01/31 06:29:18 [DEBUG] [AS3] posting request to https://172.8.1.147/mgmt/shared/appsvcs/declare/k8s
2022/01/31 06:29:18 [ERROR] [AS3] Big-IP Responded with code: 422
2022/01/31 06:29:18 [ERROR] [AS3] Raw response from Big-IP: map[code:422 message:Invalid data property: http3]
2022/01/31 06:29:18 [DEBUG] [AS3] posting request to https://172.28.15.147/mgmt/shared/appsvcs/declare/i'm_a_good_tenant
2022/01/31 06:29:18 [ERROR] [AS3] Big-IP Responded with code: 422
2022/01/31 06:29:18 [ERROR] [AS3] Raw response from Big-IP: map[code:422 message:Invalid data property: http3]
2022/01/31 06:29:18 [DEBUG] [AS3] Preparing response message to response handler for arp and fdb config
2022/01/31 06:29:18 [DEBUG] [AS3] Sent response message to response handler for arp and fdb config
2022/01/31 06:29:18 [DEBUG] [AS3] Error handling for event statusCommonResponse

```
CIS arguments

```
          args: [
            "--bigip-username=$(BIGIP_USERNAME)",
            "--bigip-password=$(BIGIP_PASSWORD)",
            "--bigip-url=https://172.8.1.147",
            "--bigip-partition=k8s",
            "--insecure=true",
            "--pool-member-type=nodeport",
            "--log-as3-response=true",
            "--manage-ingress=false",
            "--log-level=debug",
            "--manage-configmaps=true",
            "--as3-validation=true",
            #"--hubmode=true",
            "--agent=as3",
            "--filter-tenants=true",
            #"--disable-teems=true",
            "--as3-post-delay=30",
            "--verify-interval=60"
            ]
```

#### Observations (if any)

Additional query: To what length does --as3-validation=true actually validate a configmap? Is it only json syntax? schema validation?

@tkam8 CIS does only schema validation for filter-tenants or as3-validations. Data validation is taken care at AS3.
IMO, this issue is not valid based on the CIS current behaviour. 

As the feature `filter-tenants`, we are not sending 1 single AS3 declaration for all tenants, but individual declarations.

See the logs, 
Tenant: Tenant5
> 2022/01/31 06:29:18 [DEBUG] [AS3] posting request to https://172.8.1.147/mgmt/shared/appsvcs/declare/Tenant5
> 2022/01/31 06:29:18 [ERROR] [AS3] Big-IP Responded with code: 422
> 2022/01/31 06:29:18 [ERROR] [AS3] Raw response from Big-IP: map[code:422 message:Invalid data property: http3]

another AS3 post, Tenant: k8s

> 2022/01/31 06:29:18 [DEBUG] [AS3] posting request to https://172.8.1.147/mgmt/shared/appsvcs/declare/k8s
> 2022/01/31 06:29:18 [ERROR] [AS3] Big-IP Responded with code: 422
> 2022/01/31 06:29:18 [ERROR] [AS3] Raw response from Big-IP: map[code:422 message:Invalid data property: http3]

another AS3 post, Tenant:i'm_a_good_tenant

> 2022/01/31 06:29:18 [DEBUG] [AS3] posting request to https://172.28.15.147/mgmt/shared/appsvcs/declare/i'm_a_good_tenant
> 2022/01/31 06:29:18 [ERROR] [AS3] Big-IP Responded with code: 422
> 2022/01/31 06:29:18 [ERROR] [AS3] Raw response from Big-IP: map[code:422 message:Invalid data property: http3]

From all the above, CIS schema validation is passed for all the configuration and failed in data validation at AS3.

--- a/pkg/agent/as3/as3Manager.go
+++ b/pkg/agent/as3/as3Manager.go
@@ -252,7 +252,7 @@ func (am *AS3Manager) prepareTenantDeclaration(cfg *AS3Config, tenantName string
 }
 
 func (am *AS3Manager) processResponseCode(responseCode string, partition string, decl as3Declaration) {
-	if responseCode != responseStatusOk {
+	if responseCode != responseStatusOk && responseCode != responseStatusUnprocessableEntity {
 		am.failedContext.failedTenants[partition] = decl
 	} else {
 		am.excludePartitionFromFailureTenantList(partition)
@@ -290,7 +290,7 @@ func (am *AS3Manager) processTenantDeletion(tempAS3Config AS3Config) (bool, stri
 }
 
 func getResponseStatusList() map[string]int {
-	responseStatusList := map[string]int{responseStatusNotFound: 0, responseStatusServiceUnavailable: 0, responseStatusOk: 0, responseStatusCommon: 0, responseStatusDummy: 0}
+	responseStatusList := map[string]int{responseStatusNotFound: 0, responseStatusServiceUnavailable: 0, responseStatusOk: 0, responseStatusCommon: 0, responseStatusUnprocessableEntity: 0, responseStatusDummy: 0}
 	return responseStatusList
 }
 
@@ -339,6 +339,9 @@ func processResponseCodeList(responseList map[string]int) (bool, string) {
 	}
 	if responseList[responseStatusDummy] > 0 {
 		return true, responseStatusDummy
+	}
+	if responseList[responseStatusUnprocessableEntity] > 0 {
+		return true, responseStatusUnprocessableEntity
 	}
 	if responseList[responseStatusOk] > 0 {
 		return true, responseStatusOk

--- a/pkg/agent/as3/postManager.go
+++ b/pkg/agent/as3/postManager.go
@@ -39,11 +39,12 @@ const (
 )
 
 const (
-	responseStatusOk                 = "statusOK"
-	responseStatusCommon             = "statusCommonResponse"
-	responseStatusNotFound           = "statusNotFound"
-	responseStatusServiceUnavailable = "statusServiceUnavailable"
-	responseStatusDummy              = "dummy"
+	responseStatusOk                  = "statusOK"
+	responseStatusCommon              = "statusCommonResponse"
+	responseStatusNotFound            = "statusNotFound"
+	responseStatusServiceUnavailable  = "statusServiceUnavailable"
+	responseStatusUnprocessableEntity = "statusUnprocessableEntity"
+	responseStatusDummy               = "dummy"
 )
 
 type PostManager struct {
@@ -157,6 +158,8 @@ func (postMgr *PostManager) postConfigRequests(data string, url string) (bool, s
 		return postMgr.handleResponseStatusServiceUnavailable(responseMap)
 	case http.StatusNotFound:
 		return postMgr.handleResponseStatusNotFound(responseMap)
+	case http.StatusUnprocessableEntity:
+		return postMgr.handleStatusUnprocessableEntity(responseMap)
 	default:
 		return postMgr.handleResponseOthers(responseMap)
 	}
@@ -284,6 +287,19 @@ func (postMgr *PostManager) handleResponseStatusNotFound(responseMap map[string]
 		log.Errorf("[AS3] Raw response from Big-IP: %v ", responseMap)
 	}
 	return true, responseStatusNotFound
+}
+
+func (postMgr *PostManager) handleStatusUnprocessableEntity(responseMap map[string]interface{}) (bool, string) {
+	if err, ok := (responseMap["error"]).(map[string]interface{}); ok {
+		log.Errorf("[AS3] Big-IP Responded with error code: %v", err["code"])
+	} else {
+		log.Errorf("[AS3] Big-IP Responded with error code: %v", http.StatusUnprocessableEntity)
+	}
+
+	if postMgr.LogResponse {
+		log.Errorf("[AS3] Raw response from Big-IP: %v ", responseMap)
+	}
+	return true, responseStatusUnprocessableEntity
 }
 
 func (postMgr *PostManager) handleResponseOthers(responseMap map[string]interface{}) (bool, string) {

